flink-examples/flink-examples-batch/src/main/java/org/apache/flink/examples/java/clustering/KMeans.java

```diff
@@ -77,12 +80,23 @@
  *   <li>Broadcast variables in bulk iterations
  *   <li>Custom Java objects (POJOs)
  * </ul>
+ *
+ * <p>Note: All Flink DataSet APIs are deprecated since Flink 1.18 and will be removed in a future
+ * Flink major version. You can still build your application in DataSet, but you should move to
+ * either the DataStream and/or Table API. This class is retained for testing purposes.
  */
 @SuppressWarnings("serial")
 public class KMeans {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(KMeans.class);
+
     public static void main(String[] args) throws Exception {
 
+        LOGGER.warn(
+                "All Flink DataSet APIs are deprecated since Flink 1.18 and will be removed in a future"
+                        + " Flink major version. You can still build your application in DataSet, but you should move to"
+                        + " either the DataStream and/or Table API. This class is retained for testing purposes.");
```

Comment:
Can we deduplicate the warning message somewhere?

Reply:
Fixed.

Reply:
I have added a static String instant DATASET_DEPRECATION_INFO to contain the message.